# Modulo 25 Atividade 02
#### João Paulo Costa

## Cálculo

#### 1- Calcule de forma numérica a integral da função $x^3$ avaliada entre os pontos $1$ e $2$. Identifique o número de retângulos que você usou.  

In [1]:
def integral_numerica_x3(a=1.0, b=2.0, n=1000, metodo="midpoint"):
    dx = (b - a) / n
    if metodo == "left":
        xs = [a + i*dx for i in range(n)]
    elif metodo == "right":
        xs = [a + (i+1)*dx for i in range(n)]
    elif metodo == "midpoint":
        xs = [a + (i+0.5)*dx for i in range(n)]
    else:
        raise ValueError("metodo deve ser 'left', 'right' ou 'midpoint'.")

    soma = sum(x**3 for x in xs)
    return soma * dx

# Parâmetros
a, b, n = 1.0, 2.0, 1000

num_left = integral_numerica_x3(a, b, n, "left")
num_right = integral_numerica_x3(a, b, n, "right")
num_mid = integral_numerica_x3(a, b, n, "midpoint")

print(f"n = {n} retângulos")
print(f"Riemann Esquerda : {num_left:.9f}")
print(f"Riemann Direita  : {num_right:.9f}")
print(f"Ponto Médio      : {num_mid:.9f}  <= recomendado")

n = 1000 retângulos
Riemann Esquerda : 3.746500750
Riemann Direita  : 3.753500750
Ponto Médio      : 3.749999625  <= recomendado


**Resultados esperados:**

- Riemann Esquerda ≈ 3.7465
- Riemann Direita  ≈ 3.7535
- **Ponto Médio ≈ 3.75** (resultado mais preciso)

> Número de retângulos utilizados: **n = 1000**.

### 2- Calcule essa área de forma analítica.

A integral analítica é calculada como:

$$ \int_1^2 x^3 dx = \left.\frac{x^4}{4}\right|_1^2 = \frac{2^4 - 1^4}{4} = \frac{15}{4} = 3.75 $$

**Conclusão:** O valor exato da integral é **3.75**, e o cálculo numérico converge para este valor conforme aumentamos o número de retângulos.